# Preparing _Metarhizium_ data for Structure analysis

In [1]:
import pandas as pd
import seaborn as sns

/home/rodrigo/miniconda3/envs/pyviz/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
/home/rodrigo/miniconda3/envs/pyviz/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)


## Reading both datasets

In [2]:
!dir

 Volume in drive C is Windows
 Volume Serial Number is 3052-BC53

 Directory of C:\Users\OrtegaPoloRo\Documents\inglis\metarhizium\data_for_structure

10/04/2018  11:18 PM    <DIR>          .
10/04/2018  11:18 PM    <DIR>          ..
10/02/2018  11:20 AM    <DIR>          .ipynb_checkpoints
10/04/2018  11:18 PM             5,897 extraparams
10/04/2018  11:18 PM             2,006 mainparams
10/04/2018  11:18 PM            10,569 mbru_glaciation.txt
10/04/2018  11:18 PM            13,308 mbru_glaciation_str.txt_f
10/04/2018  11:18 PM            10,570 mbru_habitat.txt
10/04/2018  11:18 PM            13,293 mbru_habitat_str.txt_f
10/02/2018  06:41 PM            11,264 Mbru_Sept20_2018.txt
09/21/2018  11:43 AM            48,283 Metadata_Metarhizium_ALL_Sept21_2018.xlsx
10/04/2018  11:18 PM           162,135 Metarhizium data analysis.ipynb
10/02/2018  06:39 PM           112,173 Microsatellite_data_all_Sept20_2018.xlsx
10/04/2018  11:18 PM             1,900 mrob_glaciation.txt
10/04/2018  11

In [84]:
microsatellites = {"Mbru": 'Mbru_Sept20_2018.txt', "Mrob": 'Mrob_Sept20_2018.txt'}

In [85]:
for k,v in microsatellites.items():
    microsatellites[k] = pd.read_table(v)

In [86]:
microsatellites['Mbru'].head()

Unnamed: 0  Ma097  Ma325  Ma2049  Ma375_1  Ma375_2  Ma375_3  Ma2060  \
0        2681    178    153     133      146       -1       -1     164   
1        2687    169    163     133      146       -1       -1     164   
2        2688    169    153     133      146       -1       -1     164   
3        2689    169    153     133      146       -1       -1     164   
4        2690    178    145     129      146       -1       -1     164   

   Ma2069  Ma307_1   ...    Ma2283  Ma2296_1  Ma2296_2  Ma2077  Ma2063  \
0     212      163   ...       271       120        -1     273     140   
1     212      163   ...       271       120        -1     290     145   
2     212      163   ...       271       120        -1     290     121   
3     212      163   ...       271       120        -1     296     145   
4     221      183   ...       279       139        -1     273     145   

   Ma2292  Ma2070  Ma2056  Ma2089  Ma2054  
0     194     101     143     192     193  
1     194     101     132     192     193  
2     194     101     132     192     233  
3     194     101     137     192     233  
4     194     108     169     201     240  

[5 rows x 21 columns]

In [87]:
microsatellites['Mbru'].dtypes

Unnamed: 0    int64
Ma097         int64
Ma325         int64
Ma2049        int64
Ma375_1       int64
Ma375_2       int64
Ma375_3       int64
Ma2060        int64
Ma2069        int64
Ma307_1       int64
Ma307_2       int64
Ma2283        int64
Ma2296_1      int64
Ma2296_2      int64
Ma2077        int64
Ma2063        int64
Ma2292        int64
Ma2070        int64
Ma2056        int64
Ma2089        int64
Ma2054        int64
dtype: object

In [88]:
microsatellites['Mbru'] = microsatellites['Mbru'].rename(columns={'Unnamed: 0': 'SampleID'})

In [89]:
microsatellites['Mbru'].SampleID = microsatellites['Mbru'].SampleID.astype('category')

In [90]:
microsatellites['Mrob'].head()

SampleID  Ma097  Ma325  Ma2049  Ma375_1  Ma375_2  Ma375_3  Ma2060  Ma2069  \
0      2682    169    156     129      146       -1       -1     173     219   
1      2683    169    156     129      146       -1       -1     164     217   
2      2684    173    156     129      146       -1       -1     173     217   
3      2685    173    156     129      156       -1       -1     164     221   
4      2686    173    156     129      146       -1       -1     164     212   

   Ma307_1   ...    Ma2283  Ma2296_1  Ma2296_2  Ma2077  Ma2063  Ma2292  \
0      144   ...       266       139        -1      -1     136     194   
1      144   ...       266       139        -1      -1     136     194   
2      144   ...       266       139        -1      -1     136     194   
3      150   ...       266       139        -1      -1     136     194   
4      150   ...       266       139        -1     290     136     194   

   Ma2070  Ma2056  Ma2089  Ma2054  
0     103     137     199     200  
1     103     132     199     200  
2     230     125     199     200  
3     108     125     199     216  
4     103     147     199     216  

[5 rows x 21 columns]

In [91]:
microsatellites['Mrob'].dtypes

SampleID    int64
Ma097       int64
Ma325       int64
Ma2049      int64
Ma375_1     int64
Ma375_2     int64
Ma375_3     int64
Ma2060      int64
Ma2069      int64
Ma307_1     int64
Ma307_2     int64
Ma2283      int64
Ma2296_1    int64
Ma2296_2    int64
Ma2077      int64
Ma2063      int64
Ma2292      int64
Ma2070      int64
Ma2056      int64
Ma2089      int64
Ma2054      int64
dtype: object

In [92]:
microsatellites['Mrob'].SampleID = microsatellites['Mrob'].SampleID.astype('category')

## How many rows and columns in total?

In [93]:
microsatellites['Mbru'].shape

(134, 21)

In [94]:
microsatellites['Mrob'].shape

(25, 21)

## Filter out markers that are completely/mostly absent of all isolates

In [95]:
microsatellites['Mrob'].sum()

SampleID    69079.0
Ma097        4347.0
Ma325        3855.0
Ma2049       3218.0
Ma375_1      3532.0
Ma375_2      1491.0
Ma375_3       -25.0
Ma2060       4139.0
Ma2069       5481.0
Ma307_1      3912.0
Ma307_2       -25.0
Ma2283       6753.0
Ma2296_1     3533.0
Ma2296_2      -25.0
Ma2077       4122.0
Ma2063       3462.0
Ma2292       4858.0
Ma2070       2807.0
Ma2056       3252.0
Ma2089       4957.0
Ma2054       5429.0
dtype: float64

In [96]:
microsatellites['Mrob'].drop(['Ma375_3', 'Ma307_2', 'Ma2296_2'], axis=1, inplace=True)

In [97]:
microsatellites['Mrob'].shape

(25, 18)

In [98]:
microsatellites['Mbru'].sum()

SampleID    372841.0
Ma097        23526.0
Ma325        19636.0
Ma2049       17445.0
Ma375_1      19168.0
Ma375_2       6061.0
Ma375_3       2185.0
Ma2060       21830.0
Ma2069       29824.0
Ma307_1      23177.0
Ma307_2         73.0
Ma2283       36758.0
Ma2296_1     18838.0
Ma2296_2        35.0
Ma2077       35350.0
Ma2063       18969.0
Ma2292       25919.0
Ma2070       14400.0
Ma2056       17979.0
Ma2089       26732.0
Ma2054       30875.0
dtype: float64

### The list of column sums above identified two loci that are absent of most samples: Ma307_2 and Ma2296_2

### Let's identify how many and which samples contain two identified loci present

In [99]:
mbru_flagged = microsatellites['Mbru'].loc[:, lambda df: ['SampleID','Ma307_2', 'Ma2296_2']]

In [100]:
mbru_flagged.query('Ma307_2 != -1 | Ma2296_2 != -1')

SampleID  Ma307_2  Ma2296_2
76      2808       -1       168
115     2849      206        -1

### Only one sample per marker. Let's drop those two markers from the dataset

In [101]:
microsatellites['Mbru'] = microsatellites['Mbru'].drop(columns = ['Ma307_2', 'Ma2296_2'])

In [102]:
microsatellites['Mbru'].head()

SampleID  Ma097  Ma325  Ma2049  Ma375_1  Ma375_2  Ma375_3  Ma2060  Ma2069  \
0     2681    178    153     133      146       -1       -1     164     212   
1     2687    169    163     133      146       -1       -1     164     212   
2     2688    169    153     133      146       -1       -1     164     212   
3     2689    169    153     133      146       -1       -1     164     212   
4     2690    178    145     129      146       -1       -1     164     221   

   Ma307_1  Ma2283  Ma2296_1  Ma2077  Ma2063  Ma2292  Ma2070  Ma2056  Ma2089  \
0      163     271       120     273     140     194     101     143     192   
1      163     271       120     290     145     194     101     132     192   
2      163     271       120     290     121     194     101     132     192   
3      163     271       120     296     145     194     101     137     192   
4      183     279       139     273     145     194     108     169     201   

   Ma2054  
0     193  
1     193  
2     233  
3     233  
4     240

## Integration of metadata

### Reading metadata from the Excel file provided by Jenny Gusse

In [103]:
metadata = pd.read_excel('Metadata_Metarhizium_ALL_Sept21_2018.xlsx')

In [104]:
metadata.head()

LRC ID #  Ref Strain             Location Prov-State         Site Habitat  \
0      285           0  Abbotsford_ BC_CA 1         BC  7.1.7.A-Nat     Nat   
1      295           0  Abbotsford_ BC_CA 2         BC  7.1.7.B-Nat     Nat   
2      297           0  Abbotsford_ BC_CA 2         BC  7.1.7.B-Nat     Nat   
3      296           0  Abbotsford_ BC_CA 2         BC  7.1.7.B-Nat     Nat   
4      145           0         Agassiz 2601         BC            .      Ag   

             Ecozone Glaciation Sample ID  Soil ID Sub ID  Met by Site/Hab  \
0  Northern Cascades          I       73_8      73      8              1.0   
1  Northern Cascades          I      80_10      80     10              1.0   
2  Northern Cascades          I       80_4      80      4              1.0   
3  Northern Cascades          I       80_5      80      5              1.0   
4  Northern Cascades          I          .       .      .              1.0   

         Met ID Met Species                      Notes  
0  M brunneum B  M brunneum                        W24  
1  M brunneum A  M brunneum  W22 - Daves Roadside Spot  
2  M brunneum B  M brunneum  W22 - Daves Roadside Spot  
3  M brunneum A  M brunneum  W22 - Daves Roadside Spot  
4    M brunneum  M brunneum                          .

In [105]:
metadata.dtypes

LRC ID #            object
Ref Strain           int64
Location            object
Prov-State          object
Site                object
Habitat             object
Ecozone             object
Glaciation          object
Sample ID           object
Soil ID             object
Sub ID              object
Met by Site/Hab    float64
Met ID              object
Met Species         object
Notes               object
dtype: object

This is going to be a bit interesting. In the microsatellite data frames from above we have a __SampleID__. In the metadata file provided by Jenny, we have __LRD ID #__ and __Sample ID__, but the __SampleID__ does not seem to match the other __SampleID__ from the microsatellite data. In fact the Sample ID from the metadata is a combination of the __Soil ID__ and the __Sub ID__. That should not be an issue: the first page of the microsatellite file contains both the microsatellite SampleID and the LRC ID. I can merge those, and then merge with the metadata file using the LRC ID.

In [106]:
micro_metadata = pd.read_excel('Microsatellite_data_all_Sept20_2018.xlsx', sheet_name="All isolates")

This has been updated to reflect the _M.robertsii_ sample 2702 with LRC ID 293

In [107]:
micro_metadata = micro_metadata.rename(columns={'Sample ID #': 'SampleID'})

In [108]:
micro_metadata = micro_metadata.loc[:, lambda df: ['SampleID','LRC ID #']]

In [109]:
micro_metadata.head()

SampleID  LRC ID #
0     2681     272.0
1     2682     273.0
2     2683     274.0
3     2684     275.0
4     2685     276.0

In [110]:
micro_metadata['SampleID'] = micro_metadata['SampleID'].astype('object')
micro_metadata['LRC ID #'] = micro_metadata['LRC ID #'].astype('object')

In [111]:
meta_merge = metadata.merge(micro_metadata, how='inner', on='LRC ID #')

In [31]:
#meta_merge['SampleID'] = meta_merge['SampleID'].str.replace('(2828).*$', '\1')

In [112]:
meta_merge[meta_merge.SampleID == 2828]

LRC ID #  Ref Strain       Location Prov-State        Site Habitat  \
151      421           0  Nakusp_ BC_CA         BC  6.2.3.A-Ag      Ag   

              Ecozone Glaciation Sample ID  Soil ID Sub ID  Met by Site/Hab  \
151  Northern Rockies          I       24_3      24      3              1.0   

           Met ID Met Species                       Notes SampleID  
151  M brunneum B  M brunneum  Pasture - Clover and Grass     2828

In [113]:
meta_merge.shape

(224, 16)

In [114]:
mrob_meta = microsatellites['Mrob'].merge(meta_merge, how = 'inner', on='SampleID', indicator=True)

In [115]:
mrob_meta.shape[0] == microsatellites['Mrob'].shape[0]

True

In [116]:
mrob_meta.head()

SampleID  Ma097  Ma325  Ma2049  Ma375_1  Ma375_2  Ma2060  Ma2069  Ma307_1  \
0      2682    169    156     129      146       -1     173     219      144   
1      2683    169    156     129      146       -1     164     217      144   
2      2684    173    156     129      146       -1     173     217      144   
3      2685    173    156     129      156       -1     164     221      150   
4      2686    173    156     129      146       -1     164     212      150   

   Ma2283   ...        Ecozone  Glaciation  Sample ID   Soil ID  Sub ID  \
0     266   ...    AB Parkland           I        5_10        5      10   
1     266   ...    AB Parkland           I         5_9        5       9   
2     266   ...    AB Parkland           I         5_7        5       7   
3     266   ...    AB Parkland           I         5_8        5       8   
4     266   ...    AB Parkland           I         5_6        5       6   

   Met by Site/Hab       Met ID  Met Species  \
0              1.0  M robertsii  M robertsii   
1              1.0  M robertsii  M robertsii   
2              1.0  M robertsii  M robertsii   
3              1.0  M robertsii  M robertsii   
4              1.0  M robertsii  M robertsii   

                                 Notes  _merge  
0  Wheat field - Mature still standing    both  
1  Wheat field - Mature still standing    both  
2  Wheat field - Mature still standing    both  
3  Wheat field - Mature still standing    both  
4  Wheat field - Mature still standing    both  

[5 rows x 34 columns]

In [117]:
microsatellites['Mbru'][microsatellites['Mbru'].SampleID == 2828]

SampleID  Ma097  Ma325  Ma2049  Ma375_1  Ma375_2  Ma375_3  Ma2060  Ma2069  \
95     2828    178    145     129      146       -1       -1     164     221   

    Ma307_1  Ma2283  Ma2296_1  Ma2077  Ma2063  Ma2292  Ma2070  Ma2056  Ma2089  \
95      183     279       139     273     142     194     108     137     201   

    Ma2054  
95     243

In [118]:
mbru_meta = microsatellites['Mbru'].merge(meta_merge, how = 'left', on='SampleID', indicator=True)

In [119]:
mbru_meta

SampleID  Ma097  Ma325  Ma2049  Ma375_1  Ma375_2  Ma375_3  Ma2060  \
0        2681    178    153     133      146       -1       -1     164   
1        2687    169    163     133      146       -1       -1     164   
2        2688    169    153     133      146       -1       -1     164   
3        2689    169    153     133      146       -1       -1     164   
4        2690    178    145     129      146       -1       -1     164   
5        2691    178    145     129      146       -1       -1     164   
6        2692    178    145     129      146       -1       -1     164   
7        2693    178    145     129      146       -1       -1     164   
8        2694    178    140     129      146       -1       -1     164   
9        2695    169    145     131      146       -1       -1     164   
10       2696    178    145     129      146       -1       -1     164   
11       2697    169    153     133      146       -1       -1     164   
12       2698    178    145     129      146       -1       -1     164   
13       2699    178    163     129      146       -1       -1     152   
14       2759    178    145     129      134      150       -1     164   
15       2723    178    145     129      136      146      173     173   
16       2700    178    153     129      146       -1       -1     173   
17       2752    178    145     129      134      152       -1     164   
18       2704    178    153     137      146       -1       -1     164   
19       2705    178    153     137      146       -1       -1     164   
20       2706    178    145     129      146       -1       -1     164   
21       2707    178    145     129      146       -1       -1     164   
22       2713    178    145     129      146       -1       -1     164   
23       2714    178    145     129      136      146      164     164   
24       2715    178    145     129      136      146      164     164   
25       2806    178    145     129      146       -1       -1     173   
26       2724    169    145     131      136      146      164     164   
27       2726    178    145     129      136      146      164     164   
28       2750    178    145     129      134      152       -1     164   
29       2753    178    143     129      134      152       -1     164   
..        ...    ...    ...     ...      ...      ...      ...     ...   
104      2838    178    153     129      146       -1       -1     173   
105      2839    178    143     129      146       -1       -1     164   
106      2840    178    145     129      146       -1       -1     164   
107      2841    178    143     129      146       -1       -1     164   
108      2844    178    145     129      146       -1       -1     164   
109      2852    178    145     129      146       -1       -1     164   
110      2853    178    145     129      146       -1       -1     164   
111      2845    178    145     129      146       -1       -1     164   
112      2846    178    143     129      146       -1       -1     164   
113      2847    178    141     129      146       -1       -1     164   
114      2848    169    145     137      156       -1       -1     141   
115      2849    178    141     129      146       -1       -1     164   
116      2850    169    145     133      146       -1       -1     164   
117      2851    178    141     129      146       -1       -1     164   
118      2860    178    145     129      146       -1       -1     164   
119      2861    169    145     137      156       -1       -1     141   
120      2854    178    145     129      146       -1       -1     164   
121      2855    178    145     129      146       -1       -1     164   
122      2856    178    145     129      146       -1       -1     164   
123      2857    178    145     129      146       -1       -1     164   
124      2863    178    143     129      146       -1       -1     164   
125      2742    178    149     151      136      146      164     164   
126      

In [120]:
mbru_meta.shape[0] == microsatellites['Mbru'].shape[0]

True

In [121]:
mbru_meta.columns

Index(['SampleID', 'Ma097', 'Ma325', 'Ma2049', 'Ma375_1', 'Ma375_2', 'Ma375_3',
       'Ma2060', 'Ma2069', 'Ma307_1', 'Ma2283', 'Ma2296_1', 'Ma2077', 'Ma2063',
       'Ma2292', 'Ma2070', 'Ma2056', 'Ma2089', 'Ma2054', 'LRC ID #',
       'Ref Strain', 'Location', 'Prov-State', 'Site', 'Habitat', 'Ecozone',
       'Glaciation', 'Sample ID ', 'Soil ID', 'Sub ID', 'Met by Site/Hab',
       'Met ID', 'Met Species', 'Notes', '_merge'],
      dtype='object')

## _M. bru_ with glaciation metadata

In [122]:
mbru_glaciation = mbru_meta.loc[:, lambda df: ['SampleID','Glaciation',
                 'Ma097','Ma325', 'Ma2049', 
                 'Ma375_1', 'Ma375_2', 'Ma375_3',
                 'Ma2060', 'Ma2069', 'Ma307_1', 
                 'Ma2283', 'Ma2296_1', 'Ma2077', 
                 'Ma2063', 'Ma2292', 'Ma2070',
                 'Ma2056','Ma2089','Ma2054']]

## Replace "I" with 1 and "O" with 2

In [123]:
mbru_glaciation['Glaciation'] = mbru_glaciation['Glaciation'].str.replace("I","1")
mbru_glaciation['Glaciation'] = mbru_glaciation['Glaciation'].str.replace("O","2")

In [124]:
mbru_glaciation.tail()

SampleID Glaciation  Ma097  Ma325  Ma2049  Ma375_1  Ma375_2  Ma375_3  \
129      2703          2    178    163     129      146       -1       -1   
130      2864          2    173    151     129      136       -1       -1   
131      2865          2    173    151     129      136       -1       -1   
132      2866          2    173    155     129      146       -1       -1   
133      2867          2    173    151     129      136       -1       -1   

     Ma2060  Ma2069  Ma307_1  Ma2283  Ma2296_1  Ma2077  Ma2063  Ma2292  \
129     152     221      160     268       147     273     121     194   
130     166     221      160     266       139     275     136     194   
131     166     241      160     266       139      -1     136     194   
132     174     241      144     266       139      -1     136     194   
133     166     241      160     266       139      -1     136     194   

     Ma2070  Ma2056  Ma2089  Ma2054  
129     108     129     203     202  
130     108     132     197     218  
131     108     137     197     218  
132     108     137     197     218  
133     108     137     197     218

Interestingly, there is only one isolate that does not have metadata (2702)

In [125]:
mbru_glaciation.to_csv('mbru_glaciation.txt', sep="\t", index=False, 
                 header=['','',
                 'Ma097', 'Ma325', 'Ma2049', 
                 'Ma375_1', 'Ma375_2', 'Ma375_3',
                 'Ma2060', 'Ma2069', 'Ma307_1', 
                 'Ma2283', 'Ma2296_1', 'Ma2077', 
                 'Ma2063', 'Ma2292', 'Ma2070',
                 'Ma2056','Ma2089', 'Ma2054']
                )

## _M. bru_ with Habitat metadata

In [126]:
mbru_habitat = mbru_meta.loc[:, lambda df: ['SampleID','Habitat',
                 'Ma097','Ma325', 'Ma2049', 
                 'Ma375_1', 'Ma375_2', 'Ma375_3',
                 'Ma2060', 'Ma2069', 'Ma307_1', 
                 'Ma2283', 'Ma2296_1', 'Ma2077', 
                 'Ma2063', 'Ma2292', 'Ma2070',
                 'Ma2056','Ma2089','Ma2054']]

In [127]:
mbru_habitat.head()

SampleID Habitat  Ma097  Ma325  Ma2049  Ma375_1  Ma375_2  Ma375_3  Ma2060  \
0      2681      Ag    178    153     133      146       -1       -1     164   
1      2687      Ag    169    163     133      146       -1       -1     164   
2      2688      Ag    169    153     133      146       -1       -1     164   
3      2689      Ag    169    153     133      146       -1       -1     164   
4      2690     Nat    178    145     129      146       -1       -1     164   

   Ma2069  Ma307_1  Ma2283  Ma2296_1  Ma2077  Ma2063  Ma2292  Ma2070  Ma2056  \
0     212      163     271       120     273     140     194     101     143   
1     212      163     271       120     290     145     194     101     132   
2     212      163     271       120     290     121     194     101     132   
3     212      163     271       120     296     145     194     101     137   
4     221      183     279       139     273     145     194     108     169   

   Ma2089  Ma2054  
0     192     193  
1     192     193  
2     192     233  
3     192     233  
4     201     240

In [128]:
mbru_habitat['Habitat'] = mbru_habitat['Habitat'].str.replace("Ag","1")
mbru_habitat['Habitat'] = mbru_habitat['Habitat'].str.replace("Nat","2")

In [129]:
mbru_habitat.head()

SampleID Habitat  Ma097  Ma325  Ma2049  Ma375_1  Ma375_2  Ma375_3  Ma2060  \
0      2681       1    178    153     133      146       -1       -1     164   
1      2687       1    169    163     133      146       -1       -1     164   
2      2688       1    169    153     133      146       -1       -1     164   
3      2689       1    169    153     133      146       -1       -1     164   
4      2690       2    178    145     129      146       -1       -1     164   

   Ma2069  Ma307_1  Ma2283  Ma2296_1  Ma2077  Ma2063  Ma2292  Ma2070  Ma2056  \
0     212      163     271       120     273     140     194     101     143   
1     212      163     271       120     290     145     194     101     132   
2     212      163     271       120     290     121     194     101     132   
3     212      163     271       120     296     145     194     101     137   
4     221      183     279       139     273     145     194     108     169   

   Ma2089  Ma2054  
0     192     193  
1     192     193  
2     192     233  
3     192     233  
4     201     240

In [130]:
mbru_habitat.to_csv('mbru_habitat.txt', sep="\t", index=False, 
                 header=['','',
                 'Ma097', 'Ma325', 'Ma2049', 
                 'Ma375_1', 'Ma375_2', 'Ma375_3',
                 'Ma2060', 'Ma2069', 'Ma307_1', 
                 'Ma2283', 'Ma2296_1', 'Ma2077', 
                 'Ma2063', 'Ma2292', 'Ma2070',
                 'Ma2056','Ma2089', 'Ma2054']
                )

## _M. rob_ with glaciation data

In [131]:
mrob_meta.columns

Index(['SampleID', 'Ma097', 'Ma325', 'Ma2049', 'Ma375_1', 'Ma375_2', 'Ma2060',
       'Ma2069', 'Ma307_1', 'Ma2283', 'Ma2296_1', 'Ma2077', 'Ma2063', 'Ma2292',
       'Ma2070', 'Ma2056', 'Ma2089', 'Ma2054', 'LRC ID #', 'Ref Strain',
       'Location', 'Prov-State', 'Site', 'Habitat', 'Ecozone', 'Glaciation',
       'Sample ID ', 'Soil ID', 'Sub ID', 'Met by Site/Hab', 'Met ID',
       'Met Species', 'Notes', '_merge'],
      dtype='object')

In [132]:
mrob_glaciation = mrob_meta.loc[:, lambda df: ['SampleID','Glaciation',
                    'Ma097', 'Ma325', 'Ma2049',
                    'Ma375_1', 'Ma375_2',
                    'Ma2060', 'Ma2069', 'Ma307_1',
                    'Ma2283', 'Ma2296_1', 'Ma2077',
                    'Ma2063', 'Ma2292', 'Ma2070',
                    'Ma2056', 'Ma2089', 'Ma2054']]

In [133]:
mrob_glaciation['Glaciation'] = mrob_glaciation['Glaciation'].str.replace("I","1")
mrob_glaciation['Glaciation'] = mrob_glaciation['Glaciation'].str.replace("O","2")

After a discussion with Jenn Gusse, it was determined that SampleID 2702 is actually _M. robertsii_.

In [135]:
mrob_glaciation.loc[mbru_glaciation.index[-1]] = mrob_glaciation.iloc[-1]

In [137]:
mrob_glaciation.head()

SampleID Glaciation  Ma097  Ma325  Ma2049  Ma375_1  Ma375_2  Ma2060  \
0      2682          1    169    156     129      146       -1     173   
1      2683          1    169    156     129      146       -1     164   
2      2684          1    173    156     129      146       -1     173   
3      2685          1    173    156     129      156       -1     164   
4      2686          1    173    156     129      146       -1     164   

   Ma2069  Ma307_1  Ma2283  Ma2296_1  Ma2077  Ma2063  Ma2292  Ma2070  Ma2056  \
0     219      144     266       139      -1     136     194     103     137   
1     217      144     266       139      -1     136     194     103     132   
2     217      144     266       139      -1     136     194     230     125   
3     221      150     266       139      -1     136     194     108     125   
4     212      150     266       139     290     136     194     103     147   

   Ma2089  Ma2054  
0     199     200  
1     199     200  
2     199     200  
3     199     216  
4     199     216

In [138]:
mrob_glaciation.to_csv('mrob_glaciation.txt', sep="\t", index=False, 
                 header=['','',
                 'Ma097', 'Ma325', 'Ma2049', 
                 'Ma375_1', 'Ma375_2',
                 'Ma2060', 'Ma2069', 'Ma307_1', 
                 'Ma2283', 'Ma2296_1', 'Ma2077', 
                 'Ma2063', 'Ma2292', 'Ma2070',
                 'Ma2056','Ma2089', 'Ma2054']
                )

## _M. rob_ with habitat data

In [139]:
mrob_habitat = mrob_meta.loc[:, lambda df: ['SampleID','Habitat',
                    'Ma097', 'Ma325', 'Ma2049',
                    'Ma375_1', 'Ma375_2',
                    'Ma2060', 'Ma2069', 'Ma307_1',
                    'Ma2283', 'Ma2296_1', 'Ma2077',
                    'Ma2063', 'Ma2292', 'Ma2070',
                    'Ma2056', 'Ma2089', 'Ma2054']]

In [140]:
mrob_habitat['Habitat'] = mrob_habitat['Habitat'].str.replace("Ag","1")
mrob_habitat['Habitat'] = mrob_habitat['Habitat'].str.replace("Nat","2")

In [141]:
mrob_habitat.tail()

SampleID Habitat  Ma097  Ma325  Ma2049  Ma375_1  Ma375_2  Ma2060  Ma2069  \
20      2791       1    173    143     129      146       -1     166     212   
21      2792       1    173    145     129      146       -1     166     221   
22      2787       1    173    143     129      146       -1     166     221   
23      2799       1    169    151     131      136       -1     164     212   
24      2702       2    173    151     129      136       -1     166     221   

    Ma307_1  Ma2283  Ma2296_1  Ma2077  Ma2063  Ma2292  Ma2070  Ma2056  Ma2089  \
20      160     266       139      -1     136     194     108     137     197   
21      160     266       139     158     136     194     108     137     197   
22      160     266       139     273     136     194     108     137     197   
23      160     283       168     273     155     194     108     135     197   
24      160     266       139     273     136     194     108     125     197   

    Ma2054  
20     218  
21     218  
22     218  
23     248  
24     218

In [142]:
mrob_habitat.to_csv('mrob_habitat.txt', sep="\t", index=False, 
                 header=['','',
                 'Ma097', 'Ma325', 'Ma2049', 
                 'Ma375_1', 'Ma375_2',
                 'Ma2060', 'Ma2069', 'Ma307_1', 
                 'Ma2283', 'Ma2296_1', 'Ma2077', 
                 'Ma2063', 'Ma2292', 'Ma2070',
                 'Ma2056','Ma2089', 'Ma2054']
                )